libafl/src/bolts/minibsod.rs

```diff
@@ -0,0 +1,185 @@
+    }
+    writeln!(writer, "pc : 0x{:016x} ", ucontext.uc_mcontext.pc)?;
+
+    Ok(())
+}
+
+#[cfg(all(target_vendor = "apple", target_arch = "aarch64"))]
+fn dump_registers<W: Write>(
+    writer: &mut BufWriter<W>,
+    ucontext: &ucontext_t,
+) -> Result<(), std::io::Error> {
+    let mcontext = unsafe { *ucontext.uc_mcontext };
+    for reg in 0..29 {
+        writeln!(
+            writer,
+            "x{:02}: 0x{:016x} ",
+            reg, mcontext.__ss.__x[reg as usize]
+        );
+        if reg % 4 == 3 {
+            writeln!(writer);
+        }
+    }
+    write!(writer, "fp: 0x{:016x} ", mcontext.__ss.__fp);
+    write!(writer, "lr: 0x{:016x} ", mcontext.__ss.__lr);
+    write!(writer, "pc: 0x{:016x} ", mcontext.__ss.__pc);
+
+    Ok(())
+}
+
+#[allow(clippy::unnecessary_wraps)]
+#[cfg(all(target_vendor = "apple", target_arch = "x86_64"))]
+fn dump_registers<W: Write>(
+    _writer: &mut BufWriter<W>,
+    _ucontext: &ucontext_t,
+) -> Result<(), std::io::Error> {
+    Ok(())
+}
+
+#[cfg(all(target_os = "linux", target_arch = "x86_64"))]
+fn write_crash<W: Write>(
+    writer: &mut BufWriter<W>,
+    signal: Signal,
+    ucontext: &ucontext_t,
+) -> Result<(), std::io::Error> {
+    writeln!(
+        writer,
+        "Received signal {} at {:#016x}, fault address: {:#016x}",
+        signal,
+        ucontext.uc_mcontext.gregs[libc::REG_RIP as usize],
+        ucontext.uc_mcontext.gregs[libc::REG_CR2 as usize]
+    )?;
+
+    Ok(())
+}
+
+#[cfg(all(
+    any(target_os = "linux", target_os = "android"),
+    target_arch = "aarch64"
+))]
+fn write_crash<W: Write>(
+    writer: &mut BufWriter<W>,
+    signal: Signal,
+    ucontext: &ucontext_t,
+) -> Result<(), std::io::Error> {
+    writeln!(
+        writer,
+        "Received signal {} at 0x{:016x}, fault address: 0x{:016x}",
+        signal, ucontext.uc_mcontext.pc, ucontext.uc_mcontext.fault_address
+    )?;
+
+    Ok(())
+}
+
+#[cfg(all(target_vendor = "apple", target_arch = "aarch64"))]
+fn write_crash<W: Write>(
+    writer: &mut BufWriter<W>,
+    signal: Signal,
+    ucontext: &ucontext_t,
+) -> Result<(), std::io::Error> {
+    let mcontext = unsafe { *ucontext.uc_mcontext };
+    writeln!(
+        writer,
+        "Received signal {} at 0x{:016x}, fault address: 0x{:016x}",
+        signal, mcontext.__ss.__pc, mcontext.__es.__far
+    )?;
+
+    Ok(())
+}
+
+#[cfg(all(target_vendor = "apple", target_arch = "x86_64"))]
+fn write_crash<W: Write>(
+    writer: &mut BufWriter<W>,
+    signal: Signal,
+    _ucontext: &ucontext_t,
+) -> Result<(), std::io::Error> {
+    writeln!(writer, "Received signal {}", signal,)?;
+
+    Ok(())
+}
+
+/// Generates a mini-BSOD given a signal and context.
+#[cfg(unix)]
+pub fn generate_minibsod<W: Write>(
+    writer: &mut BufWriter<W>,
+    signal: Signal,
+    _siginfo: siginfo_t,
+    ucontext: &ucontext_t,
+) -> Result<(), std::io::Error> {
+    writeln!(writer, "{:━^100}", " CRASH ")?;
+    write_crash(writer, signal, ucontext)?;
```

Comment:
This will fail on non-linux non-apple, right?

Reply:
the whole module is `#[cfg(all(feature = "std", unix))]`...